arch/arc/core/regular_irq.S

```diff
@@ -121,7 +121,8 @@ SECTION_FUNC(TEXT, _rirq_exit)
 	 */
 	ldh_s r0, [r2, _thread_offset_to_preempt]
 	mov r3, _NON_PREEMPT_THRESHOLD
-	brhs.d r0, r3, _rirq_no_reschedule
```

Comment:
To support FPU, some instructions are added to save and load fpu related regs. This results the branch target address out of the range of brhs instruction.